src/utils.js

```diff
@@ -36,3 +64,165 @@ export const courseDateFormat = (course) => {
+            //registering parent is the same as the current student's parent
+            api.addCourseRegistration(courseToRegister(enrollment,courses[enrollment.course_id], students[enrollment.student_id]));
+            history.push("/registration/cart/");
+        } else if (!registeringParent || registeringParent === "none") {
+            api.setParentAddCourseRegistration(parentOfCurrentStudent,
+                courseToRegister(enrollment,courses[enrollment.course_id], students[enrollment.student_id]));
+            history.push("/registration/cart/");
+        }
+    };
+
+    const closeDiscardParentWarning = (toContinue) => event => {
+        event.preventDefault();
+        setDiscardParentWarning(false);
+        if (toContinue) {
+            api.setParentAddCourseRegistration(parentOfCurrentStudent,
+                courseToRegister(enrollment,courses[enrollment.course_id], students[enrollment.student_id]));
+            history.push("/registration/cart/");
+        }
+    };
+
+    const renderComponent = () => {
+        switch(componentOption){
+            case "button":{
+                return <Button
+                    onClick={handleRegisterMoreSessions}
+                    className={"button add-sessions"}
+                >
+                    Add Sessions
+                </Button>
+            }
+            case "menuItem":{
+                return <MenuItem
+                    onClick={handleRegisterMoreSessions}
+                >
+                    Add Sessions
+                </MenuItem>
+            }
+        }
+    };
+
+    return <>
+        {renderComponent()}
+        <Dialog
+            open={discardParentWarning}
+            onClose={closeDiscardParentWarning(false)}
+            aria-labelledby="warn-discard-parent"
+        >
+            <DialogTitle id="warn-discard-parent">
+                {"Finished registering parent?"}
+            </DialogTitle>
+            <DialogContent>
+                <DialogContentText>
+                    {`
+                        You are currently registering ${(registeringParent && registeringParent !== "none") && registeringParent.user.name}. If you wish to continue to add sessions, you will
+                        discard all of the currently registered courses with this parent.
+                        `}
+                </DialogContentText>
+            </DialogContent>
+            <DialogActions>
+                <Button
+                    color={"secondary"}
+                    onClick={closeDiscardParentWarning(true)}>
+                    Continue & Add Session
+                </Button>
+                <Button
+                    color={"primary"}
+                    onClick={closeDiscardParentWarning(false)}>
+                    Cancel
+                </Button>
+            </DialogActions>
+        </Dialog>
+    </>;
+};
+
+AddSessions.propTypes = {
+    componentOption: PropTypes.string.isRequired,
```

Comment:
You can use [`PropTypes.oneOf`](https://www.npmjs.com/package/prop-types#usage) to specify the exact options that this component takes.

Reply:
woaw. you learn something new every day :o